### PR TITLE
chore: error out if env init is invoked on an existing environment

### PR DESCRIPTION
**Pull request by Lou1415926**
Currently `copilot env init` twice on the same environment goes through - the operation won't fail. However, only the first invocation counts, meaning that the second invocation does not update the stack, nor the SSM parameter store.

Since `copilot env init` can only create and initialize, not update, an environment, we should error out if an update attempt is made (by running `copilot env init` twice)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/env_init.go [hunk 1]
```diff
@@ -542,6 +548,24 @@ func (o *initEnvOpts) askAdjustResources() error {
 	return nil
 }
 
+func (o *initEnvOpts) validateDuplicateEnv() error {
+	envs, err := o.store.ListEnvironments(o.appName)
```
Comment (efekarakus) [inline internal/pkg/cli/env_init.go]: Should we use `o.store.GetEnvironment()` instead? as it should be a faster endpoint

File: internal/pkg/cli/env_init.go [hunk 2]
```diff
@@ -542,6 +548,24 @@ func (o *initEnvOpts) askAdjustResources() error {
...
+	if err != nil {
+		return fmt.Errorf("validate if environment exists: %w", err)
+	}
+
+	for _, env := range envs {
+		if env.Name == o.name {
+			log.Errorf(`It seems like you are trying to init an environment that already exists.
+To re-create the environment, please run %s before running %s.
+`, color.HighlightCode(fmt.Sprintf("copilot env delete --name %s", o.name)), color.HighlightCode(fmt.Sprintf("copilot env init --name %s", o.name)))
```
Comment (efekarakus) [inline internal/pkg/cli/env_init.go]: nit:
```suggestion
			log.Errorf(`It looks like you are trying to init an environment that already exists.
To re-create the environment, please run:
1. %s 
2. And then %s
`, color.HighlightCode(fmt.Sprintf("copilot env delete --name %s", o.name)), color.HighlightCode(fmt.Sprintf("copilot env init --name %s", o.name)))
```

**Review by efekarakus — COMMENTED** [review]
Looks good! just a small suggestion

File: internal/pkg/cli/env_init.go [hunk 3]
```diff
@@ -363,6 +366,9 @@ func (o *initEnvOpts) askEnvName() error {
 		return fmt.Errorf("get environment name: %w", err)
 	}
 	o.name = envName
+	if err := o.validateDuplicateEnv(); err != nil {
+		return err
+	}
 	return nil
```
Comment (efekarakus) [inline internal/pkg/cli/env_init.go]: nit:
```suggestion
	return o.validateDuplicateEnv()
```

**Review by efekarakus — APPROVED** [review]

File: internal/pkg/cli/env_init.go [hunk 4]
```diff
@@ -542,6 +545,26 @@ func (o *initEnvOpts) askAdjustResources() error {
...
 }
 
+func (o *initEnvOpts) validateDuplicateEnv() error {
+	_, err := o.store.GetEnvironment(o.appName, o.name)
+	if err == nil {
+		log.Errorf(`It seems like you are trying to init an environment that already exists.
+To re-create the environment, please run:
```
Comment (huanjani) [inline internal/pkg/cli/env_init.go]: tiny nit:
```suggestion
To recreate the environment, please run:
```

**Review by huanjani — APPROVED** [review]
🚢

**Pull request merged**